### ISSUE TITLE
"Conflicts with committed transaction" or "Value write after transaction start" when overwriting existing keys in CassandraTransactionalKeyValue write workload

### ISSUE THREAD

**Issue by robertpang** [YugaByte/yugabyte-db#238]
When running CassandraTransactionalKeyValue write workload and after all unique keys have been written, the following errors are seen when the workload starts overwriting existing keys:

`$ java -jar ~/code/yugabyte/java/yb-loadtester/target/yb-sample-apps.jar -workload CassandraTransactionalKeyValue -num_threads_read 0 -num_threads_write 8 -nodes 127.0.0.1:9042 -num_unique_keys 2000`

`SQL error (yb/yql/cql/ql/ptree/process_context.cc:180): Invalid SQL Statement. Operation failed. Try again. (yb/common/wire_protocol.cc:194): Failed Write(tablet: 6f653374d58b4bc1849f28a559cff978, num_ops: 1, num_attempts: 1) to tablet 6f653374d58b4bc1849f28a559cff978 on tablet server be915ae366df4de18f309f50e68cb1b9 (10.150.0.28:9100) after 1 attempt(s): Conflicts with higher priority transaction: 747e31d0-49e4-4a17-b455-3b85be6a16d8`

```
2018-04-18 00:30:18,285 [INFO|com.yugabyte.sample.common.metrics.MetricsTracker|MetricsTracker] Read: 0.00 ops/sec (0.00 ms/op), 0 total ops  |  Write: 3908.64 ops/sec (8.13 ms/op), 1155859 total ops  |  Uptime: 265029 ms | maxWrittenKey: 999999 | maxGeneratedKey: 999999 |
2018-04-18 00:30:22,564 [INFO|com.yugabyte.sample.apps.AppBase|AppBase] Caught Exception:
com.datastax.driver.core.exceptions.SyntaxError: SQL error (yb/yql/cql/ql/ptree/process_context.cc:180): Invalid SQL Statement. Operation failed. Try again. (yb/common/wire_protocol.cc:194): Failed Write(tablet: 6f653374d58b4bc1849f28a559cff978, num_ops: 1, num_attempts: 1) to tablet 6f653374d58b4bc1849f28a559cff978 on tablet server be915ae366df4de18f309f50e68cb1b9 (10.150.0.28:9100) after 1 attempt(s): Conflicts with higher priority transaction: 747e31d0-49e4-4a17-b455-3b85be6a16d8
BEGIN TRANSACTION  INSERT INTO CassandraKeyValue (k, v) VALUES (:k1, :v);  INSERT INTO CassandraKeyValue (k, v) VALUES (:k2, :v);END TRANSACTION;
                                                                           ^^^^^^
 (error -11)
	at com.datastax.driver.core.exceptions.SyntaxError.copy(SyntaxError.java:58)
	at com.datastax.driver.core.exceptions.SyntaxError.copy(SyntaxError.java:24)
	at com.datastax.driver.core.DriverThrowables.propagateCause(DriverThrowables.java:37)
	at com.datastax.driver.core.DefaultResultSetFuture.getUninterruptibly(DefaultResultSetFuture.java:264)
	at com.datastax.driver.core.AbstractSession.execute(AbstractSession.java:68)
	at com.yugabyte.sample.apps.CassandraTransactionalKeyValue.doWrite(CassandraTransactionalKeyValue.java:223)
	at com.yugabyte.sample.apps.AppBase.performWrite(AppBase.java:523)
```

or `SQL error (yb/yql/cql/ql/ptree/process_context.cc:180): Invalid SQL Statement. Operation failed. Try again. (yb/common/wire_protocol.cc:194): Failed Write(tablet: afd81024dd294963b9cb55e1a1b31a49, num_ops: 1, num_attempts: 1) to tablet afd81024dd294963b9cb55e1a1b31a49 on tablet server db2fbd01f38540689a961995e0b20d55 (10.150.0.15:9100) after 1 attempt(s): Value write after transaction start`

```
2018-04-18 00:30:28,285 [INFO|com.yugabyte.sample.common.metrics.MetricsTracker|MetricsTracker] Read: 0.00 ops/sec (0.00 ms/op), 0 total ops  |  Write: 3935.28 ops/sec (8.07 ms/op), 1195355 total ops  |  Uptime: 275029 ms | maxWrittenKey: 999999 | maxGeneratedKey: 999999 |
2018-04-18 00:30:32,493 [INFO|com.yugabyte.sample.apps.AppBase|AppBase] Caught Exception:
com.datastax.driver.core.exceptions.SyntaxError: SQL error (yb/yql/cql/ql/ptree/process_context.cc:180): Invalid SQL Statement. Operation failed. Try again. (yb/common/wire_protocol.cc:194): Failed Write(tablet: afd81024dd294963b9cb55e1a1b31a49, num_ops: 1, num_attempts: 1) to tablet afd81024dd294963b9cb55e1a1b31a49 on tablet server db2fbd01f38540689a961995e0b20d55 (10.150.0.15:9100) after 1 attempt(s): Value write after transaction start
BEGIN TRANSACTION  INSERT INTO CassandraKeyValue (k, v) VALUES (:k1, :v);  INSERT INTO CassandraKeyValue (k, v) VALUES (:k2, :v);END TRANSACTION;
                                                                           ^^^^^^
 (error -11)
	at com.datastax.driver.core.exceptions.SyntaxError.copy(SyntaxError.java:58)
	at com.datastax.driver.core.exceptions.SyntaxError.copy(SyntaxError.java:24)
	at com.datastax.driver.core.DriverThrowables.propagateCause(DriverThrowables.java:37)
	at com.datastax.driver.core.DefaultResultSetFuture.getUninterruptibly(DefaultResultSetFuture.java:264)
	at com.datastax.driver.core.AbstractSession.execute(AbstractSession.java:68)
	at com.yugabyte.sample.apps.CassandraTransactionalKeyValue.doWrite(CassandraTransactionalKeyValue.java:223)
	at com.yugabyte.sample.apps.AppBase.performWrite(AppBase.java:523)
	at com.yugabyte.sample.common.IOPSThread.run(IOPSThread.java:87)
```

In debug build, this can result in a fatal:

```
F0423 17:02:29.353873 143638528 transaction.cc:228] Abort of committed transaction
Fatal failure details written to /tmp/yugabyte-local-cluster/node-1/disk-1/yb-data/tserver/logs/yb-tserver.FATAL.details.2018-04-23T17_02_29.pid79755.txt
F20180423 17:02:29 ../../../../../src/yb/client/transaction.cc:228] Abort of committed transaction
    @        0x114f44c8b  google::LogDestination::LogToSinks()
    @        0x114f43daf  google::LogMessage::SendToLog()
    @        0x114f44775  google::LogMessage::Flush()
    @        0x114f445f3  google::LogMessage::~LogMessage()
    @        0x111485bf7  yb::client::YBTransaction::Impl::Abort()
    @        0x111485a14  yb::client::YBTransaction::Abort()
    @        0x10fc2f241  yb::ql::QLEnv::Reset()
    @        0x10f3848bb  yb::ql::Executor::Reset()
    @        0x10f383e2d  yb::ql::Executor::StatementExecuted()
    @        0x10f383101  yb::ql::Executor::FlushAsyncDone()
    @        0x10f3a75d1  yb::internal::RunnableAdapter<>::Run()
    @        0x10f3a7504  yb::internal::InvokeHelper<>::MakeItSo()
    @        0x10f3a7492  yb::internal::Invoker<>::Run()
    @        0x10fc2e1c6  yb::Callback<>::Run()
    @        0x10fc2df34  yb::ql::QLEnv::ResumeCQLCall()
    @        0x10fc3aa69  yb::internal::RunnableAdapter<>::Run()
    @        0x10fc3a9bc  yb::internal::InvokeHelper<>::MakeItSo()
    @        0x10fc3a96a  yb::internal::Invoker<>::Run()
    @        0x10e4ead8e  yb::Callback<>::Run()
    @        0x10e4eacdd  yb::cqlserver::CQLInboundCall::TryResume()
    @        0x10e52122f  yb::cqlserver::CQLServiceImpl::Handle()
    @        0x113ae46e2  yb::rpc::ServicePoolImpl::Handle()
    @        0x113ae5cda  yb::rpc::(anonymous namespace)::InboundCallTask::Run()
    @        0x113ae5b6d  yb::rpc::TasksPool<>::WrappedTask::Run()
    @        0x113afa221  yb::rpc::(anonymous namespace)::Worker::Execute()
    @        0x113afb661  _ZNSt3__128__invoke_void_return_wrapperIvE6__callIJRNS_6__bindIRKMN2yb3rpc12_GLOBAL__N_16WorkerEFvvEJRKPS7_EEEEEEvDpOT_
    @        0x113afb499  _ZNSt3__110__function6__funcINS_6__bindIRKMN2yb3rpc12_GLOBAL__N_16WorkerEFvvEJRKPS6_EEENS_9allocatorISE_EEFvvEEclEv
    @        0x1106444eb  std::__1::function<>::operator()()
    @        0x1148a910f  yb::Thread::SuperviseThread()
    @     0x7fff6651b661  _pthread_body
    @     0x7fff6651b50d  _pthread_start
    @     0x7fff6651abf9  thread_start```

**Comment by mbautin** [comment]
Error message is incorrect: "Abort of committed transaction" -> "Abort of completed transaction"

**Comment by robertpang** [comment]
Fixed in commit [fee8f9d](https://github.com/YugaByte/yugabyte-db/commit/fee8f9d1c1112162fded2e636b715e5ed66fcc9b).
